tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/contact/openpaas/OpenPaasContactsConsumer.java

```diff
@@ -0,0 +1,80 @@
+package com.linagora.tmail.james.jmap.contact.openpaas;
+
+import jakarta.inject.Inject;
+
+import org.apache.james.backends.rabbitmq.ReactorRabbitMQChannelPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linagora.tmail.james.jmap.contact.TmailContactUserAddedEvent;
+
+import java.io.IOException;
+import reactor.core.Disposable;
+import scala.NotImplementedError;
+
+
+public class OpenPaasContactsConsumer {
+    private final Logger LOGGER = LoggerFactory.getLogger(OpenPaasContactsConsumer.class);
+
+    private static final String QUEUE_NAME = "contacts:contact:add";
+    private final ReactorRabbitMQChannelPool channelPool;
+    private Disposable consumeContactsDisposable;
+
+    // TODO: Create a separate RabbitMQ module for OpenPaaS communication so the injected channel pool
+    //  would be custom configured
+    @Inject
+    public OpenPaasContactsConsumer(ReactorRabbitMQChannelPool channelPool) {
+        this.channelPool = channelPool;
+    }
+
+    private TmailContactUserAddedEvent toContactUserAddedEvent(String addedContactJSONMessage) {
+        // TODO: Extract relevant contact information using JsonPath
+
+        throw new NotImplementedError();
+    }
+
+    /**
+     * Start consuming.
+     */
+    public void start() {
+        boolean doesQueueExist =
+            Boolean.TRUE.equals(channelPool.getChannelMono()
+                .map(channel -> {
+                    try {
+                        channel.queueDeclarePassive(QUEUE_NAME);
+                        return true;
+                    }
+                    catch (IOException e) {
+                        return false;
+                    }
+                }).block());
```

Comment:
That's a VERY good finding, I wasn't aware of this methods.

So if the queue already exist but with slightly different arguments does it throws? Can you please quickly test that? If no, then we have a solution to our major RabbitMQ migration nightmares Cc @QuangHoang1210 @Arsnael @vttranlina and we could use it more widely in the source code...